src/DocumentWatcher.ts

```diff
@@ -172,6 +173,22 @@ function applyEditorConfigToTextEditor(
 	/* tslint:disable:no-any */
 	textEditor.options = <any> newOptions;
 	/* tslint:enable */
+
+	applyEndOfLineSetting();
+
+	function applyEndOfLineSetting() {
+		for (const [regex, eol] of new Map([
+			[/^lf$/i, EndOfLine.LF],
+			[/^crlf$/i, EndOfLine.CRLF]
+		])) {
+			if (regex.test(editorconfig.end_of_line)) {
+				textEditor.edit(edit => {
+					edit.setEndOfLine(eol);
+				});
+				return;
+			}
+		}
```

Comment:
You can just `toLowerCase()` the setting?

```js
const endOfLine = EOL[(editorconfig.end_of_line || '').toLowerCase()];
```

I have no idea how this would work to be honest. I believe that changing the line feed in VS Code actually transform all the line endings.